### PR TITLE
Covert MediaStoreRequestHandler to Kotlin

### PR DESCRIPTION
**Pull request by amilcar-andrade**
* Also, stop using Utils.checkNotNull in favor of Kotlin's checkNotNull function

### CODE REVIEW

File: picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    return uri != null
+        && ContentResolver.SCHEME_CONTENT == uri.scheme && MediaStore.AUTHORITY == uri.authority
+  }
+
+  override fun load(picasso: Picasso, request: Request, callback: Callback) {
+    var signaledCallback = false
+    try {
+      val contentResolver = context.contentResolver
+      val requestUri = checkNotNull(request.uri, { "request.uri == null" })
+      val exifOrientation = getExifOrientation(requestUri)
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:44]:
```suggestion
      val exifOrientation = getExifOrientation(requestUri)

```

not sure why the auto-conversation all empty lines, but it's annoying!

File: picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+import android.net.Uri
+import android.provider.MediaStore
+import android.provider.MediaStore.Video
+import com.squareup.picasso3.BitmapUtils.calculateInSampleSize
+import com.squareup.picasso3.BitmapUtils.createBitmapOptions
+import com.squareup.picasso3.BitmapUtils.decodeStream
+import com.squareup.picasso3.Picasso.LoadedFrom
+
+internal class MediaStoreRequestHandler(context: Context) : ContentStreamRequestHandler(context) {
+
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:32]:
```suggestion
```

File: picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  }
+
+  override fun load(picasso: Picasso, request: Request, callback: Callback) {
+    var signaledCallback = false
+    try {
+      val contentResolver = context.contentResolver
+      val requestUri = checkNotNull(request.uri, { "request.uri == null" })
+      val exifOrientation = getExifOrientation(requestUri)
+      val mimeType = contentResolver.getType(requestUri)
+      val isVideo = mimeType != null && mimeType.startsWith("video/")
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:46]:
```suggestion
      val isVideo = mimeType != null && mimeType.startsWith("video/")

```

File: picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    val uri = data.uri
+    return uri != null
+        && ContentResolver.SCHEME_CONTENT == uri.scheme && MediaStore.AUTHORITY == uri.authority
+  }
+
+  override fun load(picasso: Picasso, request: Request, callback: Callback) {
+    var signaledCallback = false
+    try {
+      val contentResolver = context.contentResolver
+      val requestUri = checkNotNull(request.uri, { "request.uri == null" })
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:43]: i wonder what's the difference between this and `request.uri!!`?

File: picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+      if (request.hasSize()) {
+        val picassoKind = getPicassoKind(request.targetWidth, request.targetHeight)
+        if (!isVideo && picassoKind == PicassoKind.FULL) {
+          val source = getSource(requestUri)
+          val bitmap = decodeStream(source, request)
+          signaledCallback = true
+          callback.onSuccess(Result.Bitmap(bitmap, LoadedFrom.DISK, exifOrientation))
+          return
+        }
+        val id = ContentUris.parseId(requestUri)
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:56]:
```suggestion

        val id = ContentUris.parseId(requestUri)

```

File: picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+        val options = checkNotNull(createBitmapOptions(request), { "options == null" })
+        options.inJustDecodeBounds = true
+        calculateInSampleSize(
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:59]:
```suggestion

        calculateInSampleSize(
```

File: picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+          request.targetWidth, request.targetHeight, picassoKind.width,
+          picassoKind.height, options, request
+        )
+        val bitmap: Bitmap? = if (isVideo) {
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:63]:
```suggestion
        val bitmap = if (isVideo) {
```

File: picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+          Video.Thumbnails.getThumbnail(contentResolver, id, kind, options)
+        } else {
+          MediaStore.Images.Thumbnails.getThumbnail(
+            contentResolver,
+            id,
+            picassoKind.androidKind,
+            options
+          )
+        }
+        if (bitmap != null) {
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:77]:
```suggestion

        if (bitmap != null) {
```

File: picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+          signaledCallback = true
+          callback.onSuccess(Result.Bitmap(bitmap, LoadedFrom.DISK, exifOrientation))
+          return
+        }
+      }
+      val source = getSource(requestUri)
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:83]:
```suggestion

      val source = getSource(requestUri)
```

File: picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+  }
+
+  override fun getExifOrientation(uri: Uri): Int {
+    var cursor: Cursor? = null
+    return try {
+      val contentResolver = context.contentResolver
+      cursor = contentResolver.query(uri, CONTENT_ORIENTATION, null, null, null)
+      if (cursor == null || !cursor.moveToFirst()) {
+        0
+      } else cursor.getInt(0)
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:101]:
```suggestion
      } else {
        cursor.getInt(0)
      }
```

File: picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+
+  internal enum class PicassoKind(val androidKind: Int, val width: Int, val height: Int) {
+    MICRO(
+      MediaStore.Images.Thumbnails.MICRO_KIND,
+      96,
+      96
+    ),
+    MINI(MediaStore.Images.Thumbnails.MINI_KIND, 512, 384), FULL(
+      MediaStore.Images.Thumbnails.FULL_SCREEN_KIND, -1, -1
+    );
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:118]:
```suggestion
    MICRO(MediaStore.Images.Thumbnails.MICRO_KIND, 96, 96),
    MINI(MediaStore.Images.Thumbnails.MINI_KIND, 512, 384), 
    FULL(MediaStore.Images.Thumbnails.FULL_SCREEN_KIND, -1, -1);
```

geez, the auto-formatter really mangled this one...

File: picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+  }
+
+  companion object {
+    private val CONTENT_ORIENTATION = arrayOf(
+      MediaStore.Images.ImageColumns.ORIENTATION
+    )
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:124]:
```suggestion
    private val CONTENT_ORIENTATION = arrayOf(MediaStore.Images.ImageColumns.ORIENTATION)
```

**Review by jrodbx — APPROVED** [review]

File: picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt (hunk 1 above)
Comment (amilcar-andrade) [reply picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:44]: Oh true, will add spaces

File: picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt (hunk 4 above)
Comment (amilcar-andrade) [reply picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:43]: I did not want to just do `!!` given you guys were using the `Utils` stuff here, I can change it to `!!` if you prefer

**Comment by amilcar-andrade** [comment]
@jrodbx I think I addressed all your comments, except the usage of `checkNotNull`, let me know if you just want me to use `!!` was not super clear from the comment.

File: picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt (hunk 4 above)
Comment (jrodbx) [reply picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:42]: Ah, the difference is NPE vs IllegalStateException:

Screen Shot 2022-01-16 at 9 47 18 PM

Let's leave it as is, can always change it later if we change our minds.

**Pull request merged**